listenbrainz/webserver/static/js/src/RecentListens.test.tsx

```diff
@@ -43,7 +50,493 @@ const props = {
   });
 });
+
+describe("componentDidMount", () => {
+  it('calls connectWebsockets if mode is "listens" or "follow"', () => {
+    const wrapper = shallow<RecentListens>(<RecentListens {...props} />);
+    const instance = wrapper.instance();
+    instance.connectWebsockets = jest.fn();
+
+    wrapper.setState({ mode: "listens" });
+    instance.componentDidMount();
+
+    wrapper.setState({ mode: "follow" });
+    instance.componentDidMount();
+
+    expect(instance.connectWebsockets).toHaveBeenCalledTimes(2);
+  });
+
+  it('calls getRecentListensForFollowList if mode "follow"', () => {
+    const wrapper = shallow<RecentListens>(<RecentListens {...props} />);
+    const instance = wrapper.instance();
+    instance.getRecentListensForFollowList = jest.fn();
+
+    wrapper.setState({ mode: "follow", listens: [] });
+    instance.componentDidMount();
+
+    expect(instance.getRecentListensForFollowList).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe("connectWebsockets", () => {
+  it("calls createWebsocketsConnection", () => {
+    const wrapper = shallow<RecentListens>(<RecentListens {...props} />);
+    const instance = wrapper.instance();
+
+    instance.createWebsocketsConnection = jest.fn();
+    instance.connectWebsockets();
+
+    expect(instance.createWebsocketsConnection).toHaveBeenCalledTimes(1);
+  });
+
+  it("calls addWebsocketsHandlers", () => {
+    const wrapper = shallow<RecentListens>(<RecentListens {...props} />);
+    const instance = wrapper.instance();
+
+    wrapper.setState({ mode: "follow", followList: ["foo", "bar"] });
+    instance.createWebsocketsConnection();
+    // eslint-disable-next-line dot-notation
+    const spy = jest.spyOn(instance["socket"], "on");
+    instance.handleFollowUserListChange = jest.fn();
+
+    instance.addWebsocketsHandlers();
+
+    expect(spy).toHaveBeenCalled();
+    expect(instance.handleFollowUserListChange).toHaveBeenCalled();
+    expect(instance.handleFollowUserListChange).toHaveBeenCalledWith([
+      "foo",
+      "bar",
+    ]);
+  });
+
+  it('calls correct handler for "listen" event', () => {});
+
+  it('calls correct event for "playing_now" event', () => {});
+});
+
+describe("createWebsocketsConnection", () => {
+  it("calls io.connect with correct parameters", () => {
+    const wrapper = shallow<RecentListens>(<RecentListens {...props} />);
+    const instance = wrapper.instance();
+
+    const spy = jest.spyOn(io, "connect");
+    instance.connectWebsockets();
+
+    expect(spy).toHaveBeenCalledWith("http://localhost:8081");
+    jest.clearAllMocks();
+  });
+});
+
+describe("handleFollowUserListChange", () => {
+  it("sets the state correctly", () => {
+    const wrapper = shallow<RecentListens>(<RecentListens {...props} />);
+    const instance = wrapper.instance();
+
+    instance.handleFollowUserListChange(["foo", "bar"], true);
+
+    // eslint-disable-next-line dot-notation
+    expect(wrapper.state()["followList"]).toEqual(["foo", "bar"]);
```

Comment:
```suggestion
    expect(wrapper.state("followList")).toEqual(["foo", "bar"]);
```

You could do this and get rid of the eslint disable